src/types/models/mod_feedback.rs

```diff
@@ -0,0 +1,210 @@
+use std::cmp::PartialEq;
+use serde::{Serialize, Deserialize};
+use sqlx::{PgConnection, FromRow, Postgres, QueryBuilder};
+
+use crate::types::api::ApiError;
+use crate::types::models::developer::Developer;
+use crate::types::models::mod_version::ModVersion;
+
+#[derive(FromRow)]
+struct ModFeedbackRow {
+    id: i32,
+    reviewer_id: i32,
+    reviewer_name: String,
+    reviewer_admin: bool,
+    feedback_type: FeedbackTypeEnum,
+    feedback: String,
+    decision: bool,
+    dev: bool
+}
+
+#[derive(Serialize)]
+pub struct ModFeedback {
+    pub score: Score,
+    pub mod_id: String,
+    pub mod_version: String,
+    pub feedback: Vec<ModFeedbackOne>,
+}
+
+#[derive(Serialize)]
+pub struct Reviewer {
+    pub id: i32,
+    pub dev: bool,
+    pub display_name: String,
+    pub admin: bool,
+}
+
+#[derive(Serialize)]
+pub struct ModFeedbackOne {
+    pub id: i32,
+    pub reviewer: Reviewer,
+    pub feedback_type: FeedbackTypeEnum,
+    pub feedback: String,
+    pub decision: bool,
+}
+
+#[derive(Serialize)]
+pub struct Score {
+    pub score: i32,
+    pub positive: i32,
+    pub negative: i32,
+}
+
+#[derive(sqlx::Type, Serialize, Deserialize, Clone, PartialEq)]
+#[sqlx(type_name = "feedback_type")]
+pub enum FeedbackTypeEnum {
+    Positive,
+    Negative,
+    Suggestion,
+    Note
+}
+
+impl ModFeedback {
+    pub async fn get_for_mod_version_id(
+        version: &ModVersion,
+        dev_only: bool,
+        pool: &mut PgConnection,
+    ) -> Result<ModFeedback, ApiError> {
+        let mut query_builder: QueryBuilder<Postgres> = QueryBuilder::new(
+            r#"SELECT mf.id, mf.reviewer_id, dev.display_name AS reviewer_name, dev.admin AS reviewer_admin, mf.type AS "feedback_type: _", mf.feedback, mf.decision, mf.dev
+            FROM mod_feedback mf
+			INNER JOIN developers dev ON dev.id = mf.reviewer_id
+            WHERE mf.mod_version_id = "#
+        );
+        query_builder.push_bind(version.id);
+        if dev_only {
```

Comment:
i was thinking about this a little more and i think it'd be better to allow returning the current user's feedback as well, like instead of `dev_only` you have `filter_user: Option<i32>` where `None` shows all feedback and having some value returns the feedback by that user id + the dev

so it doesn't just seem like your feedback was thrown into the void

Reply:
alr